sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -331,10 +335,35 @@ class RocksDB(
     val totalSSTFilesBytes = getDBProperty("rocksdb.total-sst-files-size")
     val readerMemUsage = getDBProperty("rocksdb.estimate-table-readers-mem")
     val memTableMemUsage = getDBProperty("rocksdb.size-all-mem-tables")
-    val nativeOps = Seq("get" -> DB_GET, "put" -> DB_WRITE).toMap
-    val nativeOpsLatencyMicros = nativeOps.mapValues { typ =>
+    val nativeOpsHistograms = Seq(
+      "get" -> DB_GET,
+      "put" -> DB_WRITE,
+      "compaction" -> COMPACTION_TIME
+    ).toMap
+    val nativeOpsLatencyMicros = nativeOpsHistograms.mapValues { typ =>
       RocksDBNativeHistogram(nativeStats.getHistogramData(typ))
     }
+    val nativeOpsMetricTickers = Seq(
+      /** Number of cache misses that required reading from local disk */
+      "readBlockCacheMissCount" -> BLOCK_CACHE_MISS,
+      /** Number of cache hits that read data from RocksDB block cache avoiding local disk read */
+      "readBlockCacheHitCount" -> BLOCK_CACHE_HIT,
+      /** Number of uncompressed bytes read (from memtables/cache/sst) from DB::Get() */
+      "totalBytesReadByGet" -> BYTES_READ,
+      /** Number of uncompressed bytes issued by DB::{Put(), Delete(), Merge(), Write()} */
+      "totalBytesWrittenByPut" -> BYTES_WRITTEN,
+      /** The number of uncompressed bytes read from an iterator. */
+      "totalBytesReadThroughIterator" -> ITER_BYTES_READ,
```

Comment:
Compared to `totalBytesReadByGet`, what `totalBytesReadThroughIterator` can tell differently? Is it important to add?

Reply:
We use the iterator when processing the watermark or timeout for cleaning up the old records which reads the complete state currently. This is tracked separately from the `get` calls issued when processing the input.

Reply:
Looks like two metrics track different things.